flow/layers/image_filter_layer.cc

```diff
@@ -103,6 +113,21 @@ void ImageFilterLayer::Paint(PaintContext& context) const {
   AutoCachePaint cache_paint(context);
 
   if (context.raster_cache) {
+    // Special case an ImageFilter.matrix transformation that only
+    // applies scaling/translation. This is used by the default Android zoom
+    // page transition.
+    auto matrix_filter = filter_->asMatrix();
+    if (matrix_filter != nullptr) {
+      auto matrix = &matrix_filter->matrix();
+      auto sampling = &matrix_filter->sampling();
+      if (matrix->isScaleTranslate() && matrix->getScaleX() >= 0.0 &&
+          matrix->getScaleY() >= 0.0 &&
```

Comment:
And leaf_nodes_canvas has to be ScaleTranslate with non-negative scales.

Reply:
What is a test case I could use for this? a non-composited transform drawn in the same canvas?

Reply:
added a few...

Reply:
If either transform is not a non-negative scaletranslate then the math involved is a lot more complicated than what is written here. A non-ST parent transform means that the scales and translates in the child transform that comes from the IFL should be acting on different axes. A non-ST child matrix in the IFL would distort the axis aligned cached image box into a non-rect. Either can cause problems.

Reply:
Ahh, I used uniform scaling above so perhaps I didn't notice this.